source/base/timer.cc

```diff
@@ -353,8 +353,13 @@ TimerOutput::TimerOutput (MPI_Comm      mpi_communicator,
 
 TimerOutput::~TimerOutput()
 {
-  while (active_sections.size() > 0)
-    leave_subsection();
+  try
+    {
+      while (active_sections.size() > 0)
+        leave_subsection();
```

Comment:
It complains about three exceptions:
- `dealii::ExceptionBase` from `Assert(!active_sections.empty(),...)` in line 413
- `dealii::StandardExceptions::ExcMPI` from an `AssertThrowMPI`in `source/base/mpi.cc:225` (we call `Timer::stop` which calls `Utilities::MPI::min_max_avg`)
- `dealii::StandardExceptions::ExcMessage` from `Assert(sections.find (section_name) != sections.end(),...)`in line 420